src/Api/Auth/Controllers/WebAuthnController.cs

```diff
@@ -93,6 +114,29 @@ private async Task ValidateRequireSsoPolicyDisabledOrNotApplicable(Guid userId)
         }
     }
 
+    [HttpPut()]
+    public async Task UpdateCredential([FromBody] WebAuthnLoginCredentialUpdateRequestModel model)
+    {
+        var tokenable = _assertionOptionsDataProtector.Unprotect(model.Token);
+        if (!tokenable.TokenIsValid(WebAuthnLoginAssertionOptionsScope.PrfRegistration))
+        {
+            throw new BadRequestException("The token associated with your request is expired. A valid token is required to continue.");
+        }
+
+        var (_, credential) = await _userService.CompleteWebAuthLoginAssertionAsync(tokenable.Options, model.DeviceResponse);
+        if (credential == null)
+        {
+            throw new BadRequestException("Unable to update WebAuthnLoginCredential.");
+        }
+
+        // assign new keys to credential
+        credential.EncryptedUserKey = model.EncryptedUserKey;
+        credential.EncryptedPrivateKey = model.EncryptedPrivateKey;
+        credential.EncryptedPublicKey = model.EncryptedPublicKey;
```

Comment:
suggestion: verify this somehow? Maybe using `credential.PrfStatus === Enabled` to make sure that all keys are set and valid

Reply:
I added a check to ensure that `credential.SupportsPRf == true`, that what you mean by _valid_?

Reply:
I really meant checking `credential.PrfStatus == Enabled`, but I definitely wasn't clear on why.

The reason is that for PRF to be enabled you need all of those keys, so it doesn't really make sense to allow someone to call this endpoint without providing all of the keys. I realize now however that you might solve this issue by enforcing the fields in the api model 🤔 

Also adding `SupportsPrf === true` was actually a good idea, hadn't even crossed my mind but it doesn't really makes sense to set PRF keys for a credential that doesn't support PRF from the beginning!

Reply:
I really meant checking `credential.PrfStatus == Enabled`, but I definitely wasn't clear on why.

The reason is that for PRF to be enabled you need all of those keys, so it doesn't really make sense to allow someone to call this endpoint without providing all of the keys. I realize now however that you might solve this issue by enforcing the fields in the api model 🤔 

Also adding `SupportsPrf === true` was actually a good idea, hadn't even crossed my mind but it doesn't really makes sense to set PRF keys for a credential that doesn't support PRF from the beginning!

Reply:
I added `[Required]` attribute to the request model.